scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -529,6 +530,7 @@ class FormatOps(
         child.parent.collect {
           case _: Term.Block | _: Term.If | _: Term.While | _: Source => true
           case fun: Term.Function if isBlockFunction(fun) => true
+          case fun: Term.ContextFunction if isBlockFunction(fun) => true
```

Comment:
this is a general comment which will allow to simplify this code massively. can we instead modify scalameta to derive both Function and ContextFunction from the same trait or abstract class, with common methods and fields declared there?

Reply:
this is a common issue with scalameta api for me so far, @tgodzik is this even possible to introduce such change?

Reply:
So this is all related to macros, which are bit complex. Besides, that would be just `ContextFunction` and `Function` that could share some methods, nothing else really.

Reply:
Trying it out in https://github.com/scalameta/scalameta/pull/2207